core/src/main/java/nu/yona/server/subscriptions/entities/UserAnonymized.java

```diff
@@ -151,4 +155,13 @@ public void removeDeviceAnonymized(DeviceAnonymized deviceAnonimized)
 	{
 		return devicesAnonymized;
 	}
+
+	public void preloadGoals()
+	{
+		if (!isGoalPreloadDone)
+		{
+			Goal.getRepository().findByUserAnonymizedId(getId()); // Preload in one go
```

Comment:
Can't we inject the `GoalRepository` dependency as a parameter in the `preloadGoals` method and pass it from the `UserAnonymizedService`?